pep8ext_naming.py

```diff
@@ -165,9 +177,11 @@ class FunctionNameCheck(BaseASTCheck):
     check = LOWERCASE_REGEX.match
     N802 = "function name should be lowercase"
 
-    def visit_functiondef(self, node, parents):
+    def visit_functiondef(self, node, parents, ignore=None):
         function_type = getattr(node, 'function_type', 'function')
         name = node.name
+        if ignore and name in ignore:
```

Comment:
I'm forgetting, and this is more of a question for me, but does the AST tell you if the function definition is on a class?

I think being able to say `--ignore-names=method:setUp,method:tearDown,method:setUpClass,method:tearDownClass` would be pretty good. I'm not sure we want to allow any function defined with those names, sometimes people may only want methods.

This isn't something for this pull request, because this is good enough to satisfy the problem, but as a future enhancement that would be pretty cool.